src/packages/src/xroad/redhat/SOURCES/proxy/xroad-proxy-setup.sh

```diff
@@ -85,24 +86,30 @@ setup_database() {
     local_psql() { su -l -c "psql -qtA -p ${db_port:-5432} $*" postgres; }
     remote_psql() { psql -h "${db_addr:-127.0.0.1}" -p "${db_port:-5432}" -qtA "$@"; }
 
-    psql_dbuser() {
-        PGDATABASE="$db_database" PGUSER="$db_user" PGPASSWORD="$db_password" remote_psql "$@"
-    }
-
     if [[ -f ${root_properties} && $(get_prop ${root_properties} postgres.connection.password) != "" ]]; then
         local db_master_passwd=$(get_prop ${root_properties} postgres.connection.password)
         db_master_user=$(get_prop ${root_properties} postgres.connection.user 'postgres')
+        db_conn_master_user="${db_master_user}"
+        if [[ $(get_prop ${root_properties} postgres.connection.login_suffix) != "" ]]; then
+            suffix=$(get_prop ${root_properties} postgres.connection.login_suffix)
+            db_conn_master_user="${db_master_user}${suffix}"
+            db_conn_user="${db_user}${suffix}"
```

Comment:
On line 139, the username is (potentially) written as `user@suffix`. During subsequent upgrade, the username is read from properties (line 61), and here the suffix would be re-applied (user@suffix@suffix). And so on.
I think it would be better to read the username from the properties file, assume that it has the suffix if it is required, and then remove the suffix to get a "plain" username which is used where required.